Add CLI documentation

adriangonz opened the pull request:
## Changelog
- Add new section "Documentation / Command Line Interface", flagging it as an experimental feature.
- Generate doc command pages from `sd` (it would be good to automate this in a future).
- Set Markdown parser as `unsafe` (only for Hugo 0.6+, see https://www.docsy.dev/docs/adding-content/content/#page-contents-and-markup).

axsaucedo commented:
May be worth adding a make command to generate/update, even if it clones the other repo and then deletes it? Just thinking in case someone else wants to reproduce the creation of docs

adriangonz commented:
@axsaucedo good shout! I've added a very rough `make cli_docs` target which clones the repo, builds `sd` and then generates the doc pages.

axsaucedo commented:
Looks solid - let's add @ryandawsonuk for visibility as well

axsaucedo commented:
It may also be good to have a main page with a high level summary and perhaps the slide that we use, together with a big notice that this is "Early Access EXP", to make it clear that it's experimental so far, and that a lot of breaking changes and potentially even deprcation will be in place, as it will be quite importanat we set the expectation for this command specifically

adriangonz commented:
@axsaucedo Good suggestion! I've added the image and I've added an experimental "warning" at the top of the page. Should we make it more explicit? 

![image](https://user-images.githubusercontent.com/1577620/77054028-1a5e0880-69c7-11ea-90b0-6fa973b77939.png)

adriangonz commented:
@ryandawsonuk @gsunner @SachinVarghese is there anything else we should add for the CLI docs?
